src/runtime_src/core/edge/ps_kernels/xrt/tests/validate/ps_aie_test/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2022-2023 Advanced Micro Devices, Inc. All rights reserved.
+set(PS_KERNEL_NAME "ps_aie")
+
+add_library(${PS_KERNEL_NAME} SHARED
+  "ps_aie.cpp"
+  )
+
+set_target_properties(${PS_KERNEL_NAME} PROPERTIES 
+  VERSION ${XRT_VERSION_STRING}
+  SOVERSION ${XRT_SOVERSION}
+  )
+
+target_link_libraries(${PS_KERNEL_NAME}
+  PRIVATE
+  xrt_coreutil
+  xrt_core
+)
+
+set(XCLBIN_NAME "${PS_KERNEL_NAME}.xclbin")
+set(XCLBIN_TARGET "${PS_KERNEL_NAME}_xclbin")
+set(PS_KERNEL_LOCATION "lib${PS_KERNEL_NAME}.so")
+set(XCLBIN_LOCATION "${CMAKE_CURRENT_BINARY_DIR}/${XCLBIN_NAME}")
+
+add_custom_command(
+  OUTPUT ${XCLBIN_LOCATION}
+  COMMAND "${MY_VITIS}/bin/xclbinutil" --output ${XCLBIN_LOCATION} --add-pskernel ${PS_KERNEL_LOCATION} --force
+  DEPENDS ${PS_KERNEL_NAME}
+  )
+
+add_custom_target(${XCLBIN_TARGET} ALL
+  DEPENDS ${XCLBIN_LOCATION}
+  )
+
+install (FILES ${XCLBIN_LOCATION}
+  DESTINATION ${PS_KERNEL_INSTALL_DIR}
+  )
```

Comment:
Can all three PS kernel tests share the same CMakeLists.txt? The three CMakeLists.txt look similar except for the kernel and library name.

Reply:
We could do that. However, since each directory is making a separate library and an xclbin I would prefer to keep them in their own files. As an example all of the AIE profiling ps kernels each have their own CMakeFile. I'll see if I can create a function an reduce the repetition!